### PR TITLE
CORDA-2954: Upgrade to common-lang3

### PR DESCRIPTION
**Pull request by shamsasari**
Utilised Fields.getAllFieldsList where we were doing the equivalent

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -109,7 +109,8 @@ dependencies {
     // Apache JEXL: An embeddable expression evaluation library.
     // This may be temporary until we experiment with other ways to do on-the-fly contract specialisation via an API.
     compile "org.apache.commons:commons-jexl3:3.0"
-    compile 'commons-lang:commons-lang:2.6'
+    
+    compile "org.apache.commons:commons-lang3:3.9"
```
Comment (josecoll) [inline core/build.gradle]: Ideally defined as a variable (but ok for now as only used in this single module)

File: core/src/main/kotlin/net/corda/core/internal/StatePointerSearch.kt [hunk 2]
```diff
@@ -25,20 +26,9 @@ class StatePointerSearch(val state: ContractState) {
...
-            fields.addAll(currentClazz.declaredFields)
-            currentClazz = currentClazz.superclass
-        }
-        return fields
-    }
-
     // Helper for adding all fields to the queue.
     private fun ArrayDeque<FieldWithObject>.addAllFields(obj: Any) {
-        val fields = getAllFields(obj::class.java)
+        val fields = FieldUtils.getAllFieldsList(obj::class.java)
```
Comment (josecoll) [inline core/src/main/kotlin/net/corda/core/internal/StatePointerSearch.kt]: What Unit test coverage covers this change?

File: node/src/main/kotlin/net/corda/node/internal/Node.kt [hunk 3]
```diff
@@ -156,10 +156,9 @@ open class Node(configuration: NodeConfiguration,
 
         private fun hasMinimumJavaVersion(): Boolean {
             // when the ext.java8_minUpdateVersion gradle constant changes, so must this check
-            val major = SystemUtils.JAVA_VERSION_FLOAT
             return try {
                 val update = getJavaUpdateVersion(SystemUtils.JAVA_VERSION) // To filter out cases like 1.8.0_202-ea
-                major == 1.8F && update >= 171
+                SystemUtils.IS_JAVA_1_8 && update >= 171
```
Comment (josecoll) [inline node/src/main/kotlin/net/corda/node/internal/Node.kt]: Note: this usage goes away with JDK11 (using new internal JDK class).

File: core/src/main/kotlin/net/corda/core/internal/StatePointerSearch.kt (hunk 2 above)
Comment (shamsasari) [reply core/src/main/kotlin/net/corda/core/internal/StatePointerSearch.kt]: By default it will be covered in StatePointerSearchTests

**Pull request merged**